src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -186,15 +188,24 @@ private void RegisterTypeForMethodGen(MethodsToGen_CoreBindingHelper method, Typ
                 _sourceGenSpec.MethodsToGen_CoreBindingHelper |= method;
             }
 
-            private void RegisterTypeForBindCoreUntypedGen(TypeSpec typeSpec)
+            private void RegisterTypeForBindCoreGen(TypeSpec typeSpec)
             {
                 if (typeSpec.NeedsMemberBinding)
                 {
                     RegisterTypeForMethodGen(MethodsToGen_CoreBindingHelper.BindCore, typeSpec);
-                    RegisterTypeForMethodGen(MethodsToGen_CoreBindingHelper.BindCoreUntyped, typeSpec);
                 }
             }
 
+            private void RegisterGenMethodAsInterceptor(Enum method, IInvocationOperation operation)
+            {
+                if (!_sourceGenSpec.GenMethodsInterceptionInfo.TryGetValue(method, out List<InterceptorLocationInfo>? info))
+                {
+                    _sourceGenSpec.GenMethodsInterceptionInfo[method] = info = new List<InterceptorLocationInfo>();
```

Comment:
> _sourceGenSpec.GenMethodsInterceptionInfo[method] = info = new List<InterceptorLocationInfo>(); [](http://example.com/codeflow?start=19&length=96)

should I assume race conditions are not possible here?

Reply:
No, the generator implementation has no concurrency.